nixos/modules/services/networking/amuled.nix

```
@@ -1,89 +1,506 @@
+      PreventSleepWhileDownloading = 0;
+    };
+    "UserEvents/DownloadCompleted" = {
+      CoreEnabled = 0;
+      CoreCommand = "";
+      GUIEnabled = 0;
+      GUICommand = "";
+    };
+    "UserEvents/NewChatSession" = {
+      CoreEnabled = 0;
+      CoreCommand = "";
+      GUIEnabled = 0;
+      GUICommand = "";
+    };
+    "UserEvents/OutOfDiskSpace" = {
+      CoreEnabled = 0;
+      CoreCommand = "";
+      GUIEnabled = 0;
+      GUICommand = "";
+    };
+    "UserEvents/ErrorOnCompletion" = {
+      CoreEnabled = 0;
+      CoreCommand = "";
+      GUIEnabled = 0;
+      GUICommand = "";
+    };
+    HTTPDownload = {
+      URL_1 = "";
+    };
+  };
 
-      enable = lib.mkOption {
-        type = lib.types.bool;
-        default = false;
+  settingsOptions = {
+    eMule = {
+      Port = mkOption {
+        type = types.port;
+        default = 4662;
         description = ''
-          Whether to run the AMule daemon. You need to manually run "amuled --ec-config" to configure the service for the first time.
+          TCP port for eD2k connections.
+          Required for connecting to servers and achieving a High ID.
         '';
       };
-
-      dataDir = lib.mkOption {
-        type = lib.types.str;
-        default = "/home/${user}/";
-        defaultText = lib.literalExpression ''
-          "/home/''${config.${opt.user}}/"
+      UDPPort = mkOption {
+        type = types.port;
+        default = 4672;
+        description = ''
+          UDP port for eD2k traffic (searches, source exchange) and all Kad network communication.
+          Essential for a High ID on both networks and proper Kad functioning.
         '';
+      };
+      TempDir = mkOption {
+        type = types.path;
+        default = "${cfg.dataDir}/Temp";
+        defaultText = literalExpression "\${config.services.amule.dataDir}/Temp";
         description = ''
-          The directory holding configuration, incoming and temporary files.
+          Directory where aMule stores incomplete downloads (.part/.part.met files).
+          Ensure the aMule service has write permissions.
         '';
       };
-
-      user = lib.mkOption {
-        type = lib.types.nullOr lib.types.str;
-        default = null;
+      IncomingDir = mkOption {
+        type = types.path;
+        default = "${cfg.dataDir}/Incoming";
+        defaultText = literalExpression "\${config.services.amule.dataDir}/Incoming";
         description = ''
-          The user the AMule daemon should run as.
+          Directory where aMule moves completed downloads.
+          Files in this directory are automatically shared.
+          Ensure the aMule service has write permissions
         '';
       };
+      OSDirectory = mkOption {
+        type = types.path;
+        default = cfg.dataDir;
+        defaultText = literalExpression "\${config.services.amule.dataDir}";
+        description = "On-disk state directory, probably you don't want to change this";
+      };
```

```suggestion
      OSDirectory = mkOption {
        type = types.path;
        default = cfg.dataDir;
        defaultText = literalExpression "\${config.services.amule.dataDir}";
        description = "On-disk state directory, probably you don't want to change this.";
        internal = true;
      };
```
Options that shouldn't be changed should be marked as `internal`.